src/css/curCSS.js

```diff
@@ -1,16 +1,23 @@
 import jQuery from "../core.js";
 import isAttached from "../core/isAttached.js";
 import getStyles from "./var/getStyles.js";
+import rcustomProp from "./var/rcustomProp.js";
 
 function curCSS( elem, name, computed ) {
-	var ret;
+	var ret,
+		isCustomProp = rcustomProp.test( name );
 
 	computed = computed || getStyles( elem );
 
 	// getPropertyValue is needed for `.css('--customProperty')` (gh-3144)
 	if ( computed ) {
 		ret = computed.getPropertyValue( name ) || computed[ name ];
 
+		// trim whitespace for custom property (issue #4926)
+		if ( isCustomProp ) {
+			ret = ret.trim();
```

Comment:
@fecore1 As @gibson042 rightly noticed, we can't actually call `.trim()` as that removes all Unicode whitespace:
https://262.ecma-international.org/12.0/#sec-trimstring
https://262.ecma-international.org/12.0/#prod-WhiteSpace
We need to only remove the CSS whitespace for which we already have a regex in [`src/selector/var/whitespace.js`](https://github.com/jquery/jquery/blob/175db73ec7938e774d9e93d3afdfb35a24466b47/src/selector/var/whitespace.js), please use it.

This gotcha is a reason why we don't use `String#trim` at all in jQuery anymore but we only rely on this regex.

Reply:
wow, thanks @gibson042 , a very professional opinion, and i will fix it as soon as  possible